### PR TITLE
Add test using Jest framework

### PR DESCRIPTION
**Pull request by nechaido**
Due to request from @aqrln in #79 I open a PR for reviewing tests and choosing test framework.

### CODE REVIEW

File: test/array.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      if (item % 2 === 0) {
+        callback(new Error('Intentional error'));
+      } else {
+        callback(null, false);
+      }
+    });
+
+    metasync.find(data, predicate, (err) => {
+      expect(err).not.toBeNull();
+      expect(err).toBeInstanceOf(Error);
```
Comment (aqrln) [inline test/array.test.js]: It might be a good idea to test the error message too.

File: test/array.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const predicate = (item, callback) => process.nextTick(() => {
+      if (item % 2 === 0) {
+        callback(new Error('Intentional error'));
+      } else {
+        callback(null, false);
+      }
+    });
+
+    metasync.find(data, predicate, (err) => {
+      expect(err).not.toBeNull();
```
Comment (aqrln) [inline test/array.test.js]: Superfluous check. `null` won't pass the `instanceof` check.

File: test/array.test.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    const expected = [15];
+    const predicate = (item, callback) => process.nextTick(() => {
+      callback(null, item % 3 === 0 && item % 5 === 0);
+    });
+
+    metasync.find(data, predicate, (err, result) => {
+      expect(err).toBeNull();
+      expect(result).toBeDefined();
+      expect(result).toBeInstanceOf(Array);
+      expect(result).toEqual(expected);
```
Comment (aqrln) [inline test/array.test.js]: This check implies the previous two ones too, there's no need in them.

File: test/chain.test.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    const expected = [2, 4, 6, 8];
+    const fn = (item, callback) => process.nextTick(() => {
+      callback(null, item * 2);
+    });
+
+    metasync.for(data).map(fn).fetch((err, result) => {
+      expect(err).toBeNull();
+      expect(result).toBeDefined();
+      expect(result).toBeInstanceOf(Array);
+      expect(result).toEqual(expected);
```
Comment (aqrln) [inline test/chain.test.js]: Ditto.

File: package.json [hunk 5]
```diff
@@ -21,7 +21,7 @@
   "main": "./metasync.js",
   "readmeFilename": "README.md",
   "scripts": {
-    "test": "node tests/examples && node tests/chain && npm run lint",
+    "test": "jest && npm run lint",
```
Comment (aqrln) [inline package.json]: Yeah, and one more thing. It doesn't seem like a good idea to remove the old tests until all of them are converted. Keep them and make this `jest && node tests/examples && ...`.

**Comment by tshemsedinov** [comment]
We will use tap, #98

**Pull request closed without merging**